File: q100-q199/q100-109/q107/q107.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> ans;
        if (root == NULL) {
            return ans;
        }
        queue<TreeNode*> level_que;
        level_que.push(root);
        while (!level_que.empty()) {
            int level_size = level_que.size();
            vector<int> level;
            for (int i = 0; i < level_size; ++i) {
                TreeNode* node = level_que.front();
                level_que.pop();
                level.push_back(node->val);
                if (node->left != NULL) {
                    level_que.push(node->left);
                }
                if (node->right != NULL) {
                    level_que.push(node->right);
                }
            }
            ans.push_back(level);
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: Sequence/q100-q199/q100-109/q107/q107.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
import queue
class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        if root == None:
            return []
        ans = []
        level_que = queue.SimpleQueue()
        level_que.put(root)
        while not level_que.empty():
            level = []
            level_size = level_que.qsize()
            for i in range(0, level_size):
                node = level_que.get()
                if node.left != None:
                    level_que.put(node.left)
                if node.right != None:
                    level_que.put(node.right)
                level.append(node.val)
            ans.append(level)
        return ans[::-1]
